### DISCUSSION TITLE
Feature to specify an arbitrary textfsm template

### DISCUSSION THREAD

**Discussion by haccht** [carlmontanari/scrapli#214] [Ideas]
Hi,

I'm using `nornir_scrapli` plugin and want to specify a textfsm template for parsing a command response into structured data. `textfsm_parse_output` method only support templates from `ntc-templates` and not arbitrary ones.

I know I can do that by using `scrapli.helper.textfsm_parse` method:
```
from nornir_scrapli.tasks import send_command
from scrapli.helper import textfsm_parse

resp = nr.run(task=send_command, command=command)
structured_data = textfsm_parse('/path/to/template.textfsm', resp['router01'][0].result)
```

But I think it's more easy and intuitive this way:
```
from nornir_scrapli.tasks import send_command

resp = nr.run(task=send_command, command=command)
structured_data = resp['router01'][0].scrapli_response.textfsm_parse_output(template='/path/to/template.textfsm')
```

Thanks in advance.

**Comment by carlmontanari** [comment]
Hey @haccht 

Yeah, I think that seems like a good enhancement! I actually also want to add the ability to use a template at a URL like Michael has added to scrapligo [here](https://github.com/scrapli/scrapligo/pull/63), so this is pretty well timed. I will try to get to this this weekend, unless you are interested in putting a PR together for it?

Carl

**Comment by haccht** [comment]
Hi @carlmontanari 

Thanks for reply.
It would be nice to have a chance for contributing this awesome project. Let me give it a try for the PR! I'll try to get this weekend.
